File: ios/DeviceTypeModule.m
//
//  DeviceTypeModule.m
//  InvestecPractical
//
//  Created by Jignesh Patel on 28/09/20.
//

#import <Foundation/Foundation.h>
#import "React/RCTBridgeModule.h"

@interface RCT_EXTERN_MODULE(DeviceTypeModule, NSObject<RCTBridgeModule>)
RCT_EXTERN_METHOD(checkDeviceType:(RCTResponseSenderBlock)callback)
@end

File: ios/DeviceTypeModule.swift
//
//  DeviceTypeModule.swift
//  InvestecPractical
//
//  Created by Jignesh Patel on 28/09/20.
//

import Foundation

@objc(DeviceTypeModule)
class DeviceTypeModule: NSObject, RCTBridgeModule {
  
  @objc
  func checkDeviceType(_ callback: RCTResponseSenderBlock) {
    #if targetEnvironment(simulator)
      callback([NSNull(), "simulator"])
    #else
      callback([NSNull(), "device"])
    #endif
  }

  static func moduleName() -> String! {
    return "DeviceTypeModule"
  }
  
  static func requiresMainQueueSetup() -> Bool {
      return false
  }
}
